spring-batch-infrastructure/src/test/java/org/springframework/batch/item/data/builder/RepositoryItemReaderBuilderTests.java

```diff
@@ -0,0 +1,241 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *          http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.batch.item.data.builder;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import org.springframework.batch.item.data.RepositoryItemReader;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Sort;
+import org.springframework.data.repository.PagingAndSortingRepository;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.fail;
+import static org.mockito.Mockito.when;
+
+/**
+ * @author Glenn Renfro
+ */
+public class RepositoryItemReaderBuilderTests {
+
+	private static final String ARG1 = "foo";
+	private static final  String ARG2 = "bar";
+	private static final  String ARG3 = "baz";
+
+	public static final String TEST_CONTENT = "FOOBAR";
+
+	@Mock
+	private TestRepository repository;
+
+	@Mock
+	private Page page;
+
+	private Map<String, Sort.Direction> sorts;
+
+	private List<String> testResult;
+
+	private ArgumentCaptor<PageRequest> pageRequestContainer;
+
+	@Before
+	public void setUp() throws Exception {
+		MockitoAnnotations.initMocks(this);
+		this.sorts = new HashMap<String, Sort.Direction>();
```

Comment:
Spring Batch is java 8 now, please be sure to use diamond operators where appropriate.